src/main/frontend/app/containers/BarFormContainer.js

```diff
@@ -0,0 +1,133 @@
+import React, {Component} from 'React'
+import TextField from '../components/TextField'
+import NumberField from '../components/NumberField'
+import SelectField from '../components/SelectField'
+import TextAreaField from '../components/TextAreaField'
+
+class BarFormContainer extends Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      barName:'',
+      description:'',
+      location:'',
+      barOwner:'',
+      beachOptions:['Near the Beach','Far from the Beach'],
+      beachText:'Far from Beach',
+      hasBeach:false
+    }
+    this.handleNameInput = this.handleNameInput.bind(this)
+    this.handleDescriptionInput = this.handleDescriptionInput.bind(this)
+    this.handleLocationInput = this.handleLocationInput.bind(this)
+    this.handleBarOwnerInput = this.handleBarOwnerInput.bind(this)
+    this.handleBeachOptionsInput = this.handleBeachOptionsInput.bind(this)
+    this.handleClearForm = this.handleClearForm.bind(this)
+    this.handleSubmit = this.handleSubmit.bind(this)
+  }
+
+//handleChange function
+handleNameInput(event) {
+  this.setState({barName:event.target.value})
+}
+
+handleDescriptionInput(event) {
+  this.setState({description:event.target.value})
+}
+
+handleLocationInput(event) {
+  this.setState({location:event.target.value})
+}
+
+handleBarOwnerInput(event) {
+  this.setState({barOwner:event.target.value})
+}
+
```

Comment:
You can cut down redundancy by doing something like this ( in this example the form only handles username and password)


```
 handleChange (event) {
    // check it out: we get the event.target.name (which will be either "email" or "password")
    // and use it to target the key on our `state` object with the same name, using bracket syntax
    this.setState({ [event.target.name]: event.target.value });
  }
```